lib/linux_kernel/processor.c

```diff
@@ -644,6 +642,97 @@ M0_INTERNAL m0_processor_nr_t m0_processor_id_get(void)
 {
 	return smp_processor_id();
 }
+#elif defined CONFIG_AARCH64
+/**
+   Convert bitmap from one format to another. Copy cpumask bitmap to m0_bitmap.
+
+   @param dest -> Processors bitmap for Motr programs.
+   @param src -> Processors bitmap used by Linux kernel.
+   @param bmpsz -> Size of cpumask bitmap (src)
+
+   @pre Assumes memory is alloacted for outbmp and it's initialized.
+   
+   @see lib/processor.h
+   @see lib/bitmap.h
+ */
+static void processors_bitmap_copy(struct m0_bitmap *dest,
+				   const cpumask_t *src,
+				   uint32_t bmpsz)
+{
+	uint32_t bit;
+	bool     val;
+
+	M0_PRE(dest->b_nr >= bmpsz);
+
+	for (bit = 0; bit < bmpsz; ++bit) {
+		val = cpumask_test_cpu(bit, src);
+		m0_bitmap_set(dest, bit, val);
+	}
+}
+
+/**
+   Fetch NUMA node id for a given processor.
+
+   @param id -> id of the processor for which information is requested.
+
+   @return id of the NUMA node to which the processor belongs.
+ */
+static inline uint32_t processor_numanodeid_get(m0_processor_nr_t id)
+{
+	return cpu_to_node(id);
+}
```

Comment:
Again, code-duplication.

Reply:
even if  some code structure are  similar, Since they pertain to each processor architecture, they should be maintained respectively. It is  difficult  to iron out  the generic function however  right now

Reply:
No, they should not if they are the same code. Why difficult? You don't have to test it because the code is exactly the same.

Reply:
Fixed